### PR TITLE
Create group commands

### PR DESCRIPTION
**Pull request by julia-cp**
add Group and Grouprandom command
edit Add and Edit command to support Group

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -27,13 +28,16 @@ public class AddCommand extends Command {
...
+            + "[" + PREFIX_GROUP + "GROUP]"
             + "[" + PREFIX_TAG + "TAG]...\n"
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_NAME + "John Doe "
             + PREFIX_PHONE + "98765432 "
             + PREFIX_EMAIL + "johnd@example.com "
             + PREFIX_ADDRESS + "311, Clementi Ave 2, #02-25 "
-            + PREFIX_CATEGORY + "participant"
+            + PREFIX_CATEGORY + "participant "
+            + PREFIX_GROUP + "3 "
```
Comment (dhlee03) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:40]: Resolve the merge conflict by removing address and tag

File: src/main/java/seedu/address/logic/commands/GroupCommand.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("targetIndex", targetIndex)
+                .add("targetGroup", targetGroupNumber)
+                .toString();
+    }
+}
```
Comment (dhlee03) [inline src/main/java/seedu/address/logic/commands/GroupCommand.java:99]: Group command should be reversible command

File: src/main/java/seedu/address/logic/commands/GroupRandomCommand.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        return maxGroupSize == otherGroupCommand.maxGroupSize;
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .add("maxGroupSize", maxGroupSize)
+                .toString();
+    }
+}
```
Comment (dhlee03) [inline src/main/java/seedu/address/logic/commands/GroupRandomCommand.java:103]: similarly grouprandom can also be reversible

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 4]
```diff
@@ -32,7 +33,8 @@ public class EditCommandParser implements Parser<EditCommand> {
     public EditCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_TAG);
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
+                        PREFIX_ADDRESS,PREFIX_GROUP, PREFIX_TAG);
```
Comment (dhlee03) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:37]: remove address and tag and fix merge conflict

File: src/test/java/seedu/address/logic/commands/GroupCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
+package seedu.address.logic.commands;
+
```
Comment (dhlee03) [inline src/test/java/seedu/address/logic/commands/GroupCommandTest.java:2]: it is good to add more test cases for safety.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 6]
```diff
@@ -110,15 +120,31 @@ public CommandResult redo(Model model) throws CommandException {
...
+        if (editedPerson instanceof Staff) {
+            Staff editedStaff = (Staff) editedPerson;
+            Group updatedGroup = editPersonDescriptor.getGroup().orElse(new Group(editedStaff.getGroupNumber()));
+            editedStaff.setGroupNumber(updatedGroup.getGroupNumber());
+            return editedStaff;
+        } else if (editedPerson instanceof Participant) {
+            Participant editedParticipant = (Participant) editedPerson;
+            Group updatedGroup = editPersonDescriptor.getGroup().orElse(new Group(editedParticipant.getGroupNumber()));
+            editedParticipant.setGroupNumber(updatedGroup.getGroupNumber());
+            return editedParticipant;
```
Comment (bachletuankhai) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:142]: I think this can be done better by moving these code into Person and subclasses.

**Review by bachletuankhai — APPROVED** [review]
LGTM

**Pull request merged**